ebpfcore/ebpf_drv.c

```diff
@@ -401,6 +401,8 @@ DriverEntry(_In_ DRIVER_OBJECT* driver_object, _In_ UNICODE_STRING* registry_pat
             EBPF_TRACELOG_KEYWORD_ERROR,
             (char*)"_ebpf_driver_initialize_objects failed",
             status);
+        // Terminating ebpf trace.
+        ebpf_trace_terminate();
```

Comment:
Wouldn't this be breaking the `EBPF_LOG_EXIT()` call? I'd eventually place it after that.

```c
...
Exit:
    EBPF_LOG_EXIT();
    ebpf_trace_terminate();
    return status;
}
```

Reply:
So initially I thought that it's a bit weird to put `ebpf_trace_terminate` in the `Exit` section because isn't it weird that if `ebpf_trace_initiate` fails, a call to `ebpf_trace_terminate` still occurs?
It will work just fine because there is a check if the trace is initiated in the terminate function.
Also, putting it there will cause the trace to terminate immediately because this line will get called even in a happy flow.

Reply:
We can use a local variable to track if `ebpf_trace_initiate` succeeded and use that to determine if we need to call `ebpf_trace_terminate` in `Exit` if there was a failure.

Reply:
It's an option, but we would need to also check if the status variable is not success and only then call the terminate, otherwise we would terminate the trace even if everything was successful.
Do you want to go with this approach?